### PR TITLE
Add storybook documentation for rich text editor in composites and onPaste callback for components

### PR DESCRIPTION
**Pull request by vhuseinova-msft**
# What

- updated storybook documentation.
- Please note that inlineImages functionality will be covered in a separate task and PR

# Why

# How Tested

# Process & policy checklist

- [x] I have updated the project documentation to reflect my changes if necessary.
- [x] I have read the [CONTRIBUTING](https://github.com/Azure/communication-ui-library/blob/main/CONTRIBUTING.md) documentation.

**Is this a breaking change?**

- [ ] This change causes current functionality to break.

### CODE REVIEW

File: packages/storybook/stories/SendBox/snippets/RichTextSendBoxOnPasteCallback.snippet.tsx [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+};
+
+export const RichTextSendBoxOnPasteCallbackExample: () => JSX.Element = () => (
+  <FluentThemeProvider>
+    <div style={{ width: '31.25rem' }}>
+      <RichTextSendBox
+        onSendMessage={async () => {
+          return;
+        }}
+        onPaste={removeImageTags}
```
Comment (Leah-Xia-Microsoft) [inline packages/storybook/stories/SendBox/snippets/RichTextSendBoxOnPasteCallback.snippet.tsx:24]: The code snippet for the onPaste for both MessageThread and RTE Sendbox doesn't work when pasting plain text in.

Comment (vhuseinova-msft) [reply packages/storybook/stories/SendBox/snippets/RichTextSendBoxOnPasteCallback.snippet.tsx:24]: Could you clarify what do you mean and share steps to reproduce?

Comment (Leah-Xia-Microsoft) [reply packages/storybook/stories/SendBox/snippets/RichTextSendBoxOnPasteCallback.snippet.tsx:24]: Never mind. It was my MacOS not handling the copy paste properly.

**Comment by JoshuaLai** [comment]
I know this PR is still a WIP, just wanted to call out some of these pages have been migrated to storybook 8 so we should also make the changes over to those files as well so they are not lost

**Pull request closed without merging**